daml-lf/interface/src/main/scala/com/digitalasset/daml/lf/iface/EnvironmentInterface.scala

```diff
@@ -5,26 +5,34 @@ package com.daml.lf
 package iface
 
 import com.daml.lf.archive.Dar
-import data.Ref.Identifier
+import data.Ref.{Identifier, PackageId}
 
 import scala.collection.immutable.Map
 import scalaz.syntax.std.map._
 import scalaz.Semigroup
 
 /** The combination of multiple [[Interface]]s, such as from a dar. */
-final case class EnvironmentInterface(typeDecls: Map[Identifier, InterfaceType])
+final case class EnvironmentInterface(
+    metadata: Map[PackageId, PackageMetadata],
+    typeDecls: Map[Identifier, InterfaceType],
+)
 
 object EnvironmentInterface {
   def fromReaderInterfaces(i: Interface, o: Interface*): EnvironmentInterface =
-    EnvironmentInterface((i +: o).iterator.flatMap { case Interface(packageId, typeDecls) =>
-      typeDecls mapKeys (Identifier(packageId, _))
-    }.toMap)
+    (i +: o).foldLeft(EnvironmentInterface(Map.empty, Map.empty)) {
+      case (acc, Interface(packageId, optMetadata, typeDecls)) =>
+        acc.copy(
+          metadata =
+            optMetadata.fold(acc.metadata)(metadata => acc.metadata.updated(packageId, metadata)),
+          typeDecls = acc.typeDecls ++ typeDecls.mapKeys(Identifier(packageId, _)),
+        )
+    }
```

Comment:
I'd rather iterate over `i +: o` twice.

Reply:
why? Doing it in one go seems simple enough and should be faster

Reply:
oh didn’t realize updating maps was so much more expensive. changed 9dd1a3f91d29a03567eec4f4746632a70d37faf6

Reply:
I'm not generally opposed to updating maps iteratively, if it's clearer, but when I do, I try to use `scalaz.==>>` instead, which is guaranteed to behave like `Data.Map.Strict`.